addons/website_sale_stock/models/product_product.py

```diff
@@ -30,6 +30,9 @@ def _is_sold_out(self):
         combination_info = self.with_context(website_sale_stock_get_quantity=True).product_tmpl_id._get_combination_info(product_id=self.id)
         return combination_info['product_type'] == 'product' and combination_info['free_qty'] <= 0
 
+    def _website_show_quick_add(self):
+        return super()._website_show_quick_add() and not self._is_sold_out()
```

Comment:
I'd rather not call `super()` if the product is sold out
```suggestion
        return not self._is_sold_out() and super()._website_show_quick_add() 
```